src/log/env_config.cr

```diff
@@ -0,0 +1,17 @@
+class Log
+  # Setups *builder* based on `CRYSTAL_LOG_LEVEL` and `CRYSTAL_LOG_SOURCES`
+  # environment variables.
+  def self.setup_from_env(*, builder : Log::Builder = Log.builder,
+                          level = ENV.fetch("CRYSTAL_LOG_LEVEL", "INFO"),
+                          sources = ENV.fetch("CRYSTAL_LOG_SOURCES", ""),
+                          backend = Log::IOBackend.new)
+    builder.clear
+
+    level = Log::Severity.parse(level)
+    sources.split(',', remove_empty: false) do |source|
+      source = source.strip
+
+      builder.bind(source, level, backend)
+    end
+  end
```

Comment:
Perhaps sth like:

```suggestion
  # ...
  def self.setup(*, builder : Log::Builder = Log.builder,
                    level : Log::Severity = :info,
                    sources : Array(String) = %w[],
                    backend : Log::Backend = Log::IOBackend.new)
    builder.clear

    sources = [""] if sources.empty?
    sources.each do |source|
      builder.bind(source, level, backend)
    end
  end

  # Setups *builder* based on `CRYSTAL_LOG_LEVEL` and `CRYSTAL_LOG_SOURCES`
  # environment variables.
  def self.setup_from_env(*, builder : Log::Builder = Log.builder,
                          level_key : String = "CRYSTAL_LOG_LEVEL",
                          sources_key : String = "CRYSTAL_LOG_SOURCES",
                          backend : Log::Backend = Log::IOBackend.new)
    level = ENV.fetch(level_key, "INFO")
    level = Log::Severity.parse(level)

    sources = %w[]
    env_sources = ENV.fetch(sources_key, "")
    env_sources.split(',', remove_empty: true) do |source|
      sources << source.strip
    end

    setup(builder: builder, level: level, sources: sources, backend: backend)
  end
```

Reply:
I see hardly any benefit in separating the behaviour into two separate methods. I can only imagine the case when you want to achieve something similar as `setup_from_env` but with type safe arguments? That doesn't seem particularly more useful to the alternative. Chances are that you might need to tweak the implementation sooner or later. Anyway, this is really trivial to implement, so not a big deal to have partial duplication in application code.